Svc/DpCatalog/DpCatalog.cpp

```diff
@@ -102,10 +105,235 @@ namespace Svc {
         this->m_initialized = true;
     }
 
+    void DpCatalog::resetBinaryTree() {
```

Comment:
Is this ever called again?  I see a placement-new without a destruction first....